buildSrc/src/main/kotlin/tools-integrations-conventions.gradle.kts

```diff
@@ -61,8 +63,17 @@ configurations.all {
               req.group == "org.apache.iceberg" &&
               (req.version.isEmpty() || req.version != icebergVersionToUse)
           ) {
-            val target = "${req.group}:${req.module}:$icebergVersionToUse"
-            useTarget(target, "Managed Iceberg version to $icebergVersionToUse")
+            // TODO get rid of the internal Gradle classes DefaultImmutableVersionConstraint +
```

Comment:
Couldn't find a better way to construct new VersionConstraint + ModuleComponentSelector instances :(